tutor/specs/helpers/task-plan.spec.coffee

```diff
@@ -54,3 +56,18 @@ describe 'task-plan helper', ->
         due_at: '2015-04-05T10:00:00.000Z'
     )
     undefined
+
+
+  it 'calculates api options depending on plan isNew', ->
+    expect(TaskPlan.apiEndpointOptions(SINGLE.id, COURSE_ID))
+      .to.deep.equal(url: "plans/#{SINGLE.id}")
+
+    TaskPlanFlux.TaskPlanStore.isNew.mockReturnValue(true)
+    expect(TaskPlan.apiEndpointOptions(SINGLE.id, COURSE_ID))
+      .to.deep.equal(url: "courses/#{COURSE_ID}", method: 'POST')
+
+    TaskPlanFlux.TaskPlanStore.get.mockReturnValue(ecosystem_id: 42)
+    expect(TaskPlan.apiEndpointOptions(SINGLE.id, COURSE_ID))
+      .to.deep.equal(url: "courses/#{COURSE_ID}?ecosystem_id=42", method: 'POST')
```

Comment:
`method` doesnt seemed to be defined when not `isNew` from the helper?  which would default to `PATCH` in the api handler.